examples/toast-message/app/root.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+import {
+  json,
+  Links,
+  LiveReload,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  useLoaderData
+} from "remix";
+import type { LoaderFunction, MetaFunction } from "remix";
+
+import { Toaster, toast } from "react-hot-toast";
+import { commitSession, getSession } from "./message.server";
+
+type LoaderData = { message: string | null; type: string | null };
+
+export const meta: MetaFunction = () => {
+  return {
+    title: "Remix + Toast notifications"
+  };
+};
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("cookie"));
+
+  const message = session.get("message");
+  const type = session.get("type");
+
+  if (!message) {
+    return json<LoaderData>({ message: null, type: null });
+  }
+
+  if (!type) {
+    throw new Error("Message should have a type");
+  }
+
+  return json<LoaderData>(
+    { message, type },
+    { headers: { "Set-Cookie": await commitSession(session) } }
+  );
+};
+
+export default function App() {
+  const data = useLoaderData<LoaderData>();
```

Comment:
I would restructure this so the effect below only depend on type and message.

Reply:
if I de-structure the loader data here

```js
const {message, type} = useLoaderData()

React.useEffect(() => {
  // effect is not triggered when the message and type
  // from previous fetch is the same
}, [message, type])
```

if the value between two fetch are the same, the effect did not get triggered (eg. fetch 1 -> error, fetch 2 -> error).

but since the object will change after every time the loader is called, it triggered the effect 😅

Reply:
Let's a code comment explaining that (because a lot of people will probably bump into this.

Reply:
If you do the change to have a single flash key with both type and message of the toast, then you could destructor it here